### ISSUE TITLE
Android: QGC likes all kinds of USB devices

### ISSUE THREAD

**Issue by AndKe** [mavlink/qgroundcontrol#6139]
QGC daily for Android. 
I brought this up half a year ago, now it's too annoying to be ignored.

Then, I reported in on Note4 (and Note 10.1 2014)
Now I have a new Note 8 - and same issues right after installing QGC: 

Whenever I connect an USB adapter like logic analyzer, thermal camera, Gear VR, or almost "anything" except of USB storage, Qgroundcontrol pops up asking if I want to use it for that device. 

I know that's the OS, not QGC asking, but somehow QGC "told" Android that it's willing to handle almost any device.  (I did not check if all those devices enumerate as (among other things) an serial interface)

What I know, is that Tower, Solex, and no other software for the other devices I may connect behave that way.  

I plug in a Thermal camera , and I am offered to open it's app or QGC
I plug in an logic analyzer, offered to open it's app or QGC
I plug in an 3DR radio, and QGC is the only option.

no other apps I have respond to "everything"..

**Comment by DonLakeFlyer** [comment]
Yup. It's implemented like that on purpose. It was better that way than the reverse which is it doesn't recognize stuff. Serial on Android with a C++ application is a royal PITA as opposed to native Java.

**Comment by AndKe** [comment]
could it maybe register it's willingness to take all those devices only when QGC is running ? it's annoying for everyday use of other devices. 
Luckily "no other" app needs to behave like that, QGC should not need to do that.

**Comment by mandulaj** [comment]
Hi, is there any update to this? It was getting so annoying I had to uninstall the app. However I still want to use it. Is there no option to disable the automatic app opening? I am fine with starting the app manually when I connect telemetry to the phone.

**Comment by DonLakeFlyer** [comment]
Note that I'm only talking about the automatic launch fo QGC when you plug in a USB device. There is no way to prevent the Android prompt for which app to open (with QGC listed) when you plug in a USB device.

**Comment by khimaros** [comment]
there doesn't seem to be a way to make it remember "no"

**Comment by DonLakeFlyer** [comment]
I'll take another look at this in 5.1. Need to poke at the Tower/Solex source to see how they do it better.

**Comment by DonLakeFlyer** [comment]
@HTRamsey Can you look at this? I think usb in android is all different code now isn't it?

**Comment by DonLakeFlyer** [comment]
I'll try to take another stab at this in 5.2
